run_service.sh

```diff
@@ -156,30 +242,15 @@ then
       awk '{ print substr( $0, '$pkey_start_position', length($0) - '$pkey_start_position' ) }')
     echo "Your agent instance's autogenerated public address: $agent_address"
     echo -n "$agent_address" > "../$agent_address_path"
+    echo ""
 
     # Check balances
     agent_balance=0
     operator_balance=0
     suggested_amount=50000000000000000
-    until [[ $($PYTHON_CMD -c "print($agent_balance > ($suggested_amount-1))") == "True" && $($PYTHON_CMD -c "print($operator_balance > ($suggested_amount-1))") == "True" ]];
-    do
-        echo "Agent instance's address: $agent_address. Balance: $agent_balance WEI."
-        echo "Operator's address: $operator_address. Balance: $operator_balance WEI."
-        echo "Both of the addresses need to be funded to cover gas costs."
-        echo "Please fund them with at least 0.05 xDAI each to continue."
-        echo "Checking again in 10s..."
-        sleep 10
-        agent_balance=$(curl -s -S -X POST \
-          -H "Content-Type: application/json" \
-          --data '{"jsonrpc":"2.0","method":"eth_getBalance","params":["'"$agent_address"'","latest"],"id":1}' "$rpc" | \
-          $PYTHON_CMD -c "import sys, json; print(json.load(sys.stdin)['result'])")
-        operator_balance=$(curl -s -S -X POST \
-          -H "Content-Type: application/json" \
-          --data '{"jsonrpc":"2.0","method":"eth_getBalance","params":["'"$operator_address"'","latest"],"id":1}' "$rpc" | \
-          $PYTHON_CMD -c "import sys, json; print(json.load(sys.stdin)['result'])")
-        agent_balance=$((16#${agent_balance#??}))
-        operator_balance=$((16#${operator_balance#??}))
-    done
+
+    ensure_minimum_balance $operator_address $suggested_amount "Please, ensure the Operator's balance is at least $(wei_to_dai $suggested_amount)."
+    ensure_minimum_balance $agent_address $suggested_amount "Please, ensure your Agent instance's balance is at least $(wei_to_dai $suggested_amount)."    
```

Comment:
@Adamantios:

`agent_address` is not used for any transaction within this part of the script (minting). The balance is check below  always before running the service. 

Do you agree that we can remove this check here?

(Otherwise, on the first run of the script, the check and message appears twice, which is redundant)